android-exoplayer/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -847,14 +1094,85 @@ private WritableArray getVideoTrackInfo() {
                 videoTrack.putInt("height",format.height == Format.NO_VALUE ? 0 : format.height);
                 videoTrack.putInt("bitrate", format.bitrate == Format.NO_VALUE ? 0 : format.bitrate);
                 videoTrack.putString("codecs", format.codecs != null ? format.codecs : "");
-                videoTrack.putString("trackId",
-                        format.id == null ? String.valueOf(trackIndex) : format.id);
-                videoTracks.pushMap(videoTrack);
+                videoTrack.putString("trackId", format.id == null ? String.valueOf(trackIndex) : format.id);
+                if (isFormatSupported(format)) {
+                    videoTracks.pushMap(videoTrack);
+                }
             }
         }
+
         return videoTracks;
     }
 
+    private WritableArray getVideoTrackInfoFromManifest(Timeline timeline) {
+        return this.getVideoTrackInfoFromManifest(timeline, 0);
+    }
+
+    private WritableArray getVideoTrackInfoFromManifest(Timeline timelineRef, int retryCount) {
+        ExecutorService es = Executors.newSingleThreadExecutor();
+        final DataSource dataSource = this.mediaDataSourceFactory.createDataSource();
+        final Uri sourceUri = this.srcUri;
+        final long startTime = this.contentStartTime * 1000 - 100; // s -> ms with 100ms offset
+
+        Future<WritableArray> result = es.submit(new Callable<WritableArray>() {
+            DataSource ds = dataSource;
+            Uri uri = sourceUri;
+            Timeline timeline = timelineRef;
+            long startTimeUs = startTime * 1000; // ms -> us
+
+            public WritableArray call() throws Exception {
+                WritableArray videoTracks = Arguments.createArray();
+                try  {
+                    DashManifest manifest = DashUtil.loadManifest(this.ds, this.uri);
+                    int periodCount = manifest.getPeriodCount();
+                    for (int i = 0; i < periodCount; i++) {
+                        Period period = manifest.getPeriod(i);
```

Comment:
Is this a workaround for manifest with empty period ?

Reply:
Not sure what you mean by this? In each period we look for the video tracks and do checks like are they after the content start or supported by the device.

Reply:
Once you understand the question, let's add a comment noting this.

Reply:
Ok, DashUtil.loadManifest will download a new manifest file.
But why not using onTracksChange event ? 
This new behavior will be dash specific and it triggers not mandatory manifest download ? 
A custom manifest parser is not better ?

Reply:
Wouldn't `onTracksChange` give us the available resolutions only when content starts playing since the tracks only then become available? This was meant to show the user content resolutions even while ads are playing.

Yes, this is DASH specific, but we can expand this in the future. I updated the code to only download dash manifest if contentStartTime is set so it will no longer do unnecessary requests.

We could implement a custom manifest parser but ExoPlayer already has one so we decided to go with that since it had everything we needed to implement this feature.

Reply:
ok, you can overload the manifest parser. anyway tjis is ok, thanks